### DISCUSSION TITLE
Using Artifacts Only Server and no artifacts server

### DISCUSSION THREAD

**Discussion by guzman109** [mlflow/mlflow#11561] [Q&A]
I have two tracking servers in separate docker containers where one is in artifacts only mode and the other is not. However when I try to log a model, it does not let me and asks for credentials to my s3 bucket. 

It looks like all my requests go through the server with the no serve artifacts options. How do I make sure I can upload and download a model using the artifacts only server? I currently only set the tracking url to my no artifacts server. 

I found this but I'm still confused on what to do. Do I include it in the file name?

![image](https://github.com/mlflow/mlflow/assets/42706936/147165bf-4a69-402e-ad9d-e60e3537f2e4)

**Comment by guzman109** [comment]
Can anyone help me?

**Comment by pizzqc** [comment]
I am running into the same concerns... it isn't clear how is the client and server deployment would actually be made when running in tandem as suggested by the current documentation.

What would actually be great is to have this somewhat seamless if possible.  Here are a few options I'd like to see available:

## Option 1
Have the tracker instance referencing the artifact instance for its artifact store... similar to how the direct-access is working with S3 bucket for instance but instead of accessing the S3 bucket directly from the client you access it thru the mlflow-artifact instance which managed credentials and connection to the store.

### mlflow-tracker instance
```
args: [
              "server",
              "--dev",
              "--port", "8080",
              "--host", "0.0.0.0",
              "--no-serve-artifacts",
              "--default-artifact-root", "http://mlflow-artifact:8080",
              "--app-name", "basic-auth",
            ]
```

### mlflow-artifact instance
```
args: [
              "server",
              "--dev",
              "--port", "8080",
              "--host", "0.0.0.0",
              "--artifacts-only",              
              "--app-name", "basic-auth",
            ]
```

## Option 2
Have this implicitly configurable in the client where the client would do something like:
```
# Set up MLflow tracking URI for the tracking server
mlflow.set_tracking_uri("http://tracking-server:5000")

# Set up MLflow artifact URI for the artifact server
mlflow.set_artifact_uri("http://artifact-server:5000")
```

In a setup where MFlow server act as both, the artifact_uri would simply be the same as the one set by the set_tracking_uri


# Note
In its current form and with the available documentation,  it is simply not easy to understand how this is actually working.  I tried changing the MLFlow tracking url before calling the `log_model()` but the reality is that `log_model()` ends-up changing the status in the tracking api once the upload pass or fail.. so `log_model()` in the client is calling API that goes to both instances.

Any idea / suggestion?
